clients/lsp-rust.el

```diff
@@ -747,6 +753,7 @@ or JSON objects in `rust-project.json` format."
     :checkOnSave (:enable ,(lsp-json-bool lsp-rust-analyzer-cargo-watch-enable)
```

Comment:
> although it is internally renamed by rust-analyzer to rust-analyzer.check.allTargets which is the new version

So we also have to replace checkonsave with check right ?

Reply:
But keep this line for rust-analyzer.checkOnSave

Reply:
We don't need to rename `checkOnSave` (just yet), they do the renaming internally currently (for backward compatibility). However, eventually we probably want to go over all flags and try to follow the newest convention.

Reply:
I see, so it's probably the best solution to wait until they are happy and stop changing it =)